x-pack/legacy/plugins/infra/server/lib/adapters/framework/kibana_framework_adapter.ts

```diff
@@ -157,6 +157,8 @@ export class InfraKibanaBackendFrameworkAdapter implements InfraBackendFramework
         url = `/s/${spaceId}${url}`;
       }
     }
+    const basePath = internalRequest.getBasePath();
+    url = `${basePath}${url}`;
```

Comment:
Has this been tested with spaces? If I remember correctly, `basePath` contains the spaces part already -- see the discussion in https://github.com/elastic/kibana/pull/36765

Reply:
good point - I don't think I checked that during my review

Reply:
@skh @weltenwort that other discussion seems to imply that this shouldn't work even without spaces, if there's a base path (because `server.inject()` puts the URL after the basepath, so you'd get the basepath in the full URL twice I think?) -- if this does work without spaces, seems like we can use this instead of getting the space ID in other places in this file?

Reply:
Wasn't the cause for this bug the fact the behavior of `server.inject` changed to no include the base path since #39049? :thinking:

Reply:
In that case, it sounds like we can simplify a lot of stuff in this file 🤞@legrego from the other discussion said:

>can you use internalRequest.getBasePath() instead of deriving a space-aware URL yourself?
>
If that works, this should become something like
>
>`let url = `${internalRequest.getBasePath()}/api/metrics/vis/data`;`
>and then you can remove the check for the spaces plugin altogether.

Reply:
Verifying the spaces use-case now :)

Reply:
Fixed here: https://github.com/elastic/kibana/pull/39368/commits/b0a83146c716dd7e3fd5a84e6abbee3887cd11bf